### PR TITLE
Protect function execution in entrydb QE

### PR DESCRIPTION
**Pull request by m7onov**
Port of https://github.com/greenplum-db/gpdb/pull/15096 to main

### CODE REVIEW

File: src/backend/executor/functions.c [hunk 1]
```diff
@@ -229,7 +229,8 @@ querytree_safe_for_qe_walker(Node *expr, void *context)
 						if (!(IsCatalogNamespace(namespaceId) ||
 									IsToastNamespace(namespaceId) ||
 									IsAoSegmentNamespace(namespaceId) ||
-									IsReplicatedTable(rte->relid)))
+									(IsReplicatedTable(rte->relid) &&
+                   !IS_QUERY_DISPATCHER())))
```
Comment (kainwen) [inline src/backend/executor/functions.c:233]: The format is a bit strange. Put it in the same line?

**Review by kainwen — APPROVED** [review]

File: src/backend/executor/functions.c (hunk 1 above)
Comment (m7onov) [reply src/backend/executor/functions.c]: Ooops. Fixed now.

**Comment by kainwen** [comment]
Looks good. Would you please squash into 1 commit and write better commit message? Then I will push this PR.

Thanks a lot!

**Comment by m7onov** [comment]
Done. Is it okay now?

**Pull request merged**